index.js

```diff
@@ -637,10 +666,47 @@ SendStream.prototype.sendFile = function sendFile(path) {
     fs.stat(p, function (err, stat) {
       if (err) return next(err)
       if (stat.isDirectory()) return next()
-      self.emit('file', p, stat)
-      self.send(p, stat)
+      checkPrecompressionAndSendFile(p, stat)
     })
   }
+
+  function checkPrecompressionAndSendFile(p, stat) {
+    self.emit('file', p, stat)
+    if (!self._precompressionFormats) return self.send(p, stat)
+
+    var state = {
+      contents: [],
+      extensionsToCheck: self._precompressionFormats.length
+    }
+
+    self._precompressionFormats.forEach(function (format) {
+      debug('stat "%s%s"', p, format.extension);
+      fs.stat(p + format.extension, function onstat(err, contentStat) {
```

Comment:
Why bother stating files we don't intend to send to the client? For example, this will stat for .gz and .br, even if the client only accepts gzip.